dcpy/connectors/edm/publishing.py

```diff
@@ -200,50 +228,104 @@ def legacy_upload(
             s3.copy_file(BUCKET, str(key), str(prefix / "latest" / output.name), acl)
 
 
+def promote_to_draft(
+    build_key: BuildKey,
+    acl: s3.ACL,
+    keep_build: bool = True,
+    max_files: int = s3.MAX_FILE_COUNT,
+    draft_revision_summary: str = "",
+):
+    version = get_version(build_key)
+
+    # generate version draft revision number
+    draft_revision_number = (
+        len(get_draft_version_revisions(build_key.product, version)) + 1
+    )
+    draft_revision_label = versions.DraftVersionRevision(
+        draft_revision_number, draft_revision_summary
+    ).label
+
+    # read in build metadata, update it with draft label, and save it locally
+    build_metadata = get_build_metadata(product_key=build_key)
+    build_metadata.draft_revision_name = draft_revision_label
+    build_metadata_path = Path("build_metadata.json")
+    with open(build_metadata_path, "w", encoding="utf-8") as f:
+        json.dump(build_metadata.model_dump(), f, indent=4)
+
+    # promote from build to draft
+    source = build_key.path + "/"
+    target = f"{build_key.product}/draft/{version}/{draft_revision_label}/"
+    s3.copy_folder(BUCKET, source, target, acl, max_files=max_files)
+
+    # upload updated metadata file
+    s3.upload_file(
+        BUCKET,
+        build_metadata_path,
+        f"{target}{build_metadata_path.name}",
+        acl,
+    )
+
+    logger.info(
+        f"Promoted {build_key.product} to drafts as {version}/{draft_revision_label}"
+    )
+
+    if not keep_build:
+        s3.delete(BUCKET, source)
+
+
+def determine_publish_version(
```

Comment:
Bit of a nit, but just pondering the name of this function. It is determining, but I wouldn't call it the most intuitive name. Like maybe `validate_or_patch_version` or something like that? Not that we should go with that specifically, just food for thought